npm/sparrowdb/index.d.ts does not parse as TypeScript (144 syntax errors at HEAD)

Found while adding bindings in #447. Not caused by it.

## Problem

The checked-in, published type declarations do not parse. `tsc --noEmit --skipLibCheck` against `npm/sparrowdb/index.d.ts` at `main` reports **144 syntax errors**, the first at line 32:

```
index.d.ts(32,2): error TS1109: Expression expected.
index.d.ts(32,8): error TS1005: ';' expected.
index.d.ts(33,2): error TS1109: Expression expected.
...
```

Reproduce:

```bash
npx typescript@5 tsc --noEmit --skipLibCheck npm/sparrowdb/index.d.ts
```

## Cause

Several Rust doc comments in `crates/sparrowdb-node/src/lib.rs` illustrate the generated TypeScript using *nested* doc comments. NAPI-RS copies them verbatim into a JSDoc block, and the inner `*/` closes the outer block early — everything after it is parsed as code. From the generated file:

```typescript
/**
 * A single result row from a vector or full-text search.
 *
 * ```typescript
 * interface NodeResult {
 *   /** Node ID as a decimal string (full u64 range, JS-safe). */   <-- closes the outer comment here
 *   id: string;
 *   ...
```

`NodeResult` is the first occurrence; there are more further down.

## Impact

Any TypeScript consumer importing `sparrowdb` gets a broken declaration file. It degrades silently — `skipLibCheck` is on by default in many setups, which suppresses *checking* the file but not the parse errors, and editors surface it as missing/incorrect types rather than as a broken package.

## Fix

In `crates/sparrowdb-node/src/lib.rs`, replace nested `/** ... */` inside the ```typescript examples with `//` line comments. #447 does this for the one struct it adds (`VectorIndexHealth`) and, as a side effect of re-syncing the comment nesting, brings the count from 144 to 35 — the remaining 35 are the other occurrences.

Worth adding a CI step so it cannot regress:

```bash
npx tsc --noEmit npm/sparrowdb/index.d.ts
```

Note the file is regenerated by `napi build`, so the fix belongs in the Rust doc comments, not in the `.d.ts`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Aq3C5J646sBCH7sVs3uaCE

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -200,6 +200,12 @@ jobs:
         with:
           node-version: '20'
 
+      - name: Type-check npm/sparrowdb/index.d.ts
+        working-directory: npm/sparrowdb
+        run: |
+          npm install
+          npm run typecheck
+
       - name: Build sparrowdb-node (.node binary)
         run: |
           cargo build --release -p sparrowdb-node

--- a/crates/sparrowdb-node/src/lib.rs
+++ b/crates/sparrowdb-node/src/lib.rs
@@ -241,9 +241,9 @@ pub struct QueryResult {
 ///
 /// ```typescript
 /// interface NodeResult {
-///   /** Node ID as a decimal string (full u64 range, JS-safe). */
+///   // Node ID as a decimal string (full u64 range, JS-safe).
 ///   id: string;
-///   /** Relevance score: cosine similarity / BM25 score / distance. */
+///   // Relevance score: cosine similarity / BM25 score / distance.
 ///   score: number;
 /// }
 /// ```
@@ -267,9 +267,12 @@ pub struct NodeResult {
 /// }
 /// ```
 ///
-/// Note the `//` comments above rather than nested doc comments: a `*/` inside
-/// a generated JSDoc block closes it early, which is why the checked-in
-/// `npm/sparrowdb/index.d.ts` does not currently parse as TypeScript.
+/// Note the `//` comments above rather than nested doc comments. A JSDoc
+/// close-marker inside a generated JSDoc block terminates it early, and every
+/// declaration after that point is then parsed as code — which is exactly how
+/// the checked-in `npm/sparrowdb/index.d.ts` stopped parsing as TypeScript
+/// (#449). `npm run typecheck` in CI now guards against a recurrence; keep
+/// examples in these doc comments to `//` so it never fires.
 #[napi(object)]
 pub struct VectorIndexHealth {
     /// Vectors stored in the index.

--- a/npm/sparrowdb/index.d.ts
+++ b/npm/sparrowdb/index.d.ts
@@ -28,8 +28,10 @@ export interface QueryResult {
  *
  * ```typescript
  * interface NodeResult {
- *   id: string;   // Node ID as a decimal string (full u64 range, JS-safe).
- *   score: number; // Relevance score: cosine similarity / BM25 score / distance.
+ *   // Node ID as a decimal string (full u64 range, JS-safe).
+ *   id: string;
+ *   // Relevance score: cosine similarity / BM25 score / distance.
+ *   score: number;
  * }
  * ```
  */
@@ -50,9 +52,12 @@ export interface NodeResult {
  * }
  * ```
  *
- * Note the `//` comments above rather than nested doc comments: a `*\/` inside
- * a generated JSDoc block closes it early, which is why the checked-in
- * `npm/sparrowdb/index.d.ts` does not currently parse as TypeScript.
+ * Note the `//` comments above rather than nested doc comments. A JSDoc
+ * close-marker inside a generated JSDoc block terminates it early, and every
+ * declaration after that point is then parsed as code — which is exactly how
+ * the checked-in `npm/sparrowdb/index.d.ts` stopped parsing as TypeScript
+ * (#449). `npm run typecheck` in CI now guards against a recurrence; keep
+ * examples in these doc comments to `//` so it never fires.
  */
 export interface VectorIndexHealth {
   /** Vectors stored in the index. */

--- a/npm/sparrowdb/index.d.ts
+++ b/npm/sparrowdb/index.d.ts
@@ -28,10 +28,8 @@ export interface QueryResult {
  *
  * ```typescript
  * interface NodeResult {
- *   /** Node ID as a decimal string (full u64 range, JS-safe). */
- *   id: string;
- *   /** Relevance score: cosine similarity / BM25 score / distance. */
- *   score: number;
+ *   id: string;   // Node ID as a decimal string (full u64 range, JS-safe).
+ *   score: number; // Relevance score: cosine similarity / BM25 score / distance.
  * }
  * ```
  */

--- a/npm/sparrowdb/index.js
+++ b/npm/sparrowdb/index.js
@@ -58,4 +58,17 @@ function loadNative() {
   )
 }
 
-module.exports = loadNative()
+const native = loadNative()
+
+module.exports = native
+
+// Named exports must be assigned explicitly (not just spread onto
+// `module.exports`) so that Node's ESM/CJS interop — which relies on static
+// analysis via cjs-module-lexer, not evaluation — can see them. A dynamic
+// `module.exports = loadNative()` alone is invisible to that analysis, so an
+// ESM consumer's `import { SparrowDB } from 'sparrowdb'` fails with "Named
+// export 'SparrowDB' not found" even though `require('sparrowdb').SparrowDB`
+// works fine under CommonJS. See issue #449.
+module.exports.SparrowDB = native.SparrowDB
+module.exports.ReadTx = native.ReadTx
+module.exports.WriteTx = native.WriteTx

--- a/npm/sparrowdb/package.json
+++ b/npm/sparrowdb/package.json
@@ -4,13 +4,21 @@
   "description": "Native Node.js/TypeScript bindings for SparrowDB — embedded graph database with Cypher query support",
   "main": "index.js",
   "types": "index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./index.d.ts",
+      "default": "./index.js"
+    },
+    "./package.json": "./package.json"
+  },
   "files": [
     "index.js",
     "index.d.ts",
     "*.node"
   ],
   "scripts": {
-    "test": "node --test test/integration.test.js"
+    "test": "node --test test/integration.test.js",
+    "typecheck": "tsc -p tsconfig.json"
   },
   "engines": {
     "node": ">= 18"
@@ -29,6 +37,7 @@
     "@sparrowdb/win32-x64-msvc": "0.1.25"
   },
   "devDependencies": {
-    "@napi-rs/cli": "^2.18.4"
+    "@napi-rs/cli": "^2.18.4",
+    "typescript": "^5.6.0"
   }
 }

--- a/npm/sparrowdb/test/consumer.ts
+++ b/npm/sparrowdb/test/consumer.ts
@@ -1,0 +1,43 @@
+// Type-check smoke test for the published `.d.ts` — not executed, only
+// compiled. Exercises the shapes real TypeScript consumers rely on: named
+// import of the class exports, `execute`, `executeWithParams`, and iterating
+// `QueryResult.rows`. CI type-checks this file against `index.d.ts` so a
+// change to the declarations that breaks a realistic consumer fails the
+// build instead of shipping silently (see issue #449).
+import { SparrowDB, ReadTx, WriteTx } from 'sparrowdb'
+
+// Stand-in for a real sink (console.log) so this file type-checks without
+// pulling in @types/node just to know the shape of `console`.
+declare function use(value: unknown): void
+
+const db: SparrowDB = SparrowDB.open('/tmp/example.db')
+
+const result = db.execute('MATCH (n:Person) RETURN n.name LIMIT 5')
+for (const row of result.rows) {
+  use(row['n.name'])
+}
+
+const paramResult = db.executeWithParams(
+  'MERGE (k:Memory {id: $id}) SET k.embedding = $emb',
+  { id: 'abc-123', emb: [0.1, 0.2, 0.3] },
+)
+use(paramResult.columns)
+
+const health = db.vectorIndexHealth('Memory', 'embedding')
+use([health.stored, health.reachable, health.unreachable])
+
+const hits = db.hybridSearch('Memory', 'content', 'embedding', new Float32Array(768), 'query', 10, 0.7)
+for (const hit of hits) {
+  use([hit.id, hit.score])
+}
+
+const readTx: ReadTx = db.beginRead()
+use(readTx.snapshotTxnId)
+
+const writeTx: WriteTx = db.beginWrite()
+writeTx.execute("CREATE (:Person {name: 'Ada'})")
+const newTxnId: string = writeTx.commit()
+use(newTxnId)
+
+db.checkpoint()
+db.optimize()

--- a/npm/sparrowdb/tsconfig.json
+++ b/npm/sparrowdb/tsconfig.json
@@ -1,0 +1,17 @@
+{
+  "compilerOptions": {
+    "strict": true,
+    "target": "ES2020",
+    "lib": ["ES2020"],
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "types": [],
+    "noEmit": true,
+    "skipLibCheck": false,
+    "baseUrl": ".",
+    "paths": {
+      "sparrowdb": ["./index.d.ts"]
+    }
+  },
+  "include": ["index.d.ts", "test/consumer.ts"]
+}

